core/src/main/java/hudson/model/Computer.java

```diff
@@ -828,7 +828,19 @@ protected void kill() {
     protected void onRemoved(){
     }
 
-    private synchronized void setNumExecutors(int n) {
+    /**
+     * Calling path, *means protected by Queue.withLock
+     *
+     * Computer.doConfigSubmit -> Computer.replaceBy ->Jenkins.setNodes* ->Computer.setNode
+     * AbstractCIBase.updateComputerList->Computer.inflictMortalWound*
+     * AbstractCIBase.updateComputerList->AbstractCIBase.updateComputer* ->Computer.setNode
+     * AbstractCIBase.updateComputerList->AbstractCIBase.killComputer->Computer.kill
+     * Computer.constructor->Computer.setNode
+     * Computer.kill is called after numExecutors set to zero(Computer.inflictMortalWound) so not need the Queue.lock
+     *
+     * @param number of executors
+     */
+    private void setNumExecutors(int n) {
```

Comment:
Can we add a `@GuardedBy("hudson.model.Queue.lock")` annotation. Hopefully that way findbugs can confirm your analysis (if it does, we would need to suppress FB warnings on `Computer.kill`) and that way we can protect against further changes breaking the protection